src/app/painel-administrativo/(admin-routes)/produto/[[...slug]]/page.tsx

```diff
@@ -1,15 +1,42 @@
 'use client';
 
-import { Container } from '../../produtos/styles';
+import { Container } from './styles';
+
+import ProductPage from '@/app/components/ProductPage';
+
+import { useSession } from 'next-auth/react';
 
 type ProductProps = {
   params: {
     slug: string;
+    productId: string;
   };
+  businessId: number | string;
+  mode?: 'private' | 'public';
 };
 
 export default function Product(props: ProductProps) {
-  const mode = props?.params?.slug ? 'edit' : 'register';
+  const { data: session } = useSession();
+  const businessId = session?.data.business[0].id;
+  const modePage = props?.params?.slug ? 'edit' : 'register';
+
+  console.log(modePage);
+
+  //Categories request
 
-  return <Container>mode {mode}</Container>;
+  return (
+    <Container>
+      {!!businessId ? (
+        <ProductPage
+          modePage={modePage}
+          businessId={businessId}
+          productId={props?.params?.slug}
+        />
+      ) : (
+        <div>
+          <h1>Crie alguma categoria de alimentos antes de criar um produto.</h1>
```

Comment:
Aqui podemos ter um problema, pois não é feito somente para alimentos.

```suggestion
          <h1>Crie alguma categoria antes de criar um produto.</h1>
```